lib/std/fs/path.zig

```diff
@@ -749,8 +749,18 @@ fn testResolvePosix(paths: []const []const u8, expected: []const u8) !void {
     return testing.expect(mem.eql(u8, actual, expected));
 }
 
+/// Returns the parant path of a directory path, or null when the root is reached.
+/// Asserts that the path has a directory component.
+pub fn parent(path: []const u8) ?[]const u8 {
+    const new_path = dirname(path) orelse unreachable;
+    if (new_path.len == path.len) return null;
```

Comment:
There are a few edge cases that should be handled, such as `dirname('//') = '/'` that the length check doesn't detect.

Reply:
The length check does exactly check for this case to happen:
When passing in `/` or `C:\`, it has no parent component, so the path returned has the exact same length as the one passed in, in all other cases, the length will shorten and thus, a parent directory is found

Reply:
`"//".len != "/".len`, the check fails yet the two strings point to the same path and they're both root dirs.

Reply:
Yeah, I also think we shouldn't handle this edge case in here. Lets say someone is calling `parent` with `////` then the first time `parent` returns `/` and then the next time it will find the root (because of `dirname` returning `/` for `/`). Which seems fine in my opinion.

Reply:
So you're telling me that the parent of the root is the root itself?

> then the next time it will find the root 

You're assuming this is repeatedly applied, nobody is stopping me from calling it just once and get the wrong result.

[Rust](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9d767c2dc1681033e05105d86b7fe083) shares my view and returns `None` when applied to `"//"`.

Reply:
Hmm I see. Okay, let me see how to implement this.

Reply:
`//` is *not* equivalent to `/`, infact in posix, `//` is allowed to point to an entirely different root/tree.
However, three or more slashes (`///`) can indeed be simplified to `/`

Reply:
Alright, I stand corrected. Posix is so full of edge cases... AFAICS the double slash has no meaning on Linux/BSD but under Cygwin it's used to refer to network shares.
If we go full-posix then `dirname //` should return `//` no?

Reply:
>  If we go full-posix then `dirname //` should return `//` no?

correct. and `dirname ///` could return `/`